BraveWallet/Crypto/Stores/NetworkStore.swift

```diff
@@ -44,6 +44,88 @@ public class NetworkStore: ObservableObject {
     }
     rpcService.add(self)
   }
+  
+  public func updateSelectedNetwork(_ network: BraveWallet.EthereumChain) {
+    if selectedChainId != network.chainId {
+      rpcService.setNetwork(network.chainId) { [self] success in
+        if success {
+          selectedChainId = network.chainId
+        }
+      }
+    }
+  }
+  
+  // MARK: - Custom Networks
+  
+  @Published var isAddingNewNetwork: Bool = false
+  
+  public func addCustomNetwork(_ network: BraveWallet.EthereumChain,
+                               completion: @escaping (_ accpted: Bool) -> Void) {
+    func addNetwors(_ network: BraveWallet.EthereumChain, completion: @escaping (_ accpted: Bool) -> Void) {
+      rpcService.add(network) { [self] chainId, status, message in
+        if status == .success {
+          // meaning removal is also succeeded if id exists. need to update the list by removing the old then adding the new
+          if let index = ethereumChains.firstIndex(where: { $0.id.lowercased() == network.id.lowercased() }) {
+            ethereumChains.remove(at: index)
+          }
```

Comment:
Should we not just fetch networks and overwrite `ethereumChains` from the `RpcService` instead of manually removing/appending the local variable?